### PR TITLE
Check trace and proxy chain ids

### PR DESCRIPTION
**Pull request by aomerk**

### CODE REVIEW

File: inspect/scorecalc/pass_fail_calculator.go [hunk 1]
```diff
@@ -48,6 +48,16 @@ func (c *chainPassFailCalculator) CalculateScore(results *inspect.InspectionResu
 		return 0, nil
 	}
 
+	// proxy should point to correct chain id
+	if results.Indicators[inspect.IndicatorProxyAPIChainID] != float64(c.config.ChainID) {
+		return 0, nil
+	}
+
+	// trace api should point to correct chain id
+	if results.Indicators[inspect.IndicatorTraceAPIChainID] != float64(c.config.ChainID) {
```
Comment (stevenlanders) [inline inspect/scorecalc/pass_fail_calculator.go:57]: Is this safe for chains that don't have trace? (-1 != 56)?

Comment (aomerk) [reply inspect/scorecalc/pass_fail_calculator.go]: great catch, 94dc12e8ada147c812ccdaea8de2c540d7017309 should fix it

File: inspect/network.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func (ni *NetworkInspector) Inspect(ctx context.Context, inspectionCfg Inspectio
 }
 
 // speedTest returns download speed, upload speed and error.
-func speedTest() (float64, float64, error) {
+func _() (float64, float64, error) {
```
Comment (canercidam) [inline inspect/network.go:68]: Does the linter error go away if we rename to `SpeedTest` instead? That could be a little more useful.

File: inspect/api.go [hunk 3]
```diff
@@ -53,10 +53,16 @@ func getRpcResponse(ctx context.Context, rpcClient *rpc.Client, respData interfa
 func GetNetworkID(ctx context.Context, rpcClient *rpc.Client) (*big.Int, error) {
 	var result string
```
Comment (canercidam) [inline inspect/api.go:54]: Can we change this to `resultStr` and then `resultHex` to `result` instead? The second one looks like we are trying a general fallback by leveraging `(*big.Int).MarshalText`.

**Pull request merged**